ClosedXML/Excel/CalcEngine/Functions/Text.cs

```diff
@@ -276,6 +277,72 @@ private static object _Text(List<Expression> p)
                 return nf.Format(number, CultureInfo.InvariantCulture);
         }
 
+        /// <summary>
+        /// A function to Join text https://support.office.com/en-us/article/textjoin-function-357b449a-ec91-49d0-80c3-0e8fc845691c
+        /// </summary>
+        /// <param name="p">Parameters</param>
+        /// <returns> string </returns>
+        /// <exception cref="ApplicationException">
+        /// Delimiter in first param must be a string
+        /// or
+        /// Second param must be a boolen (TRUE/FALSE)
+        /// </exception>
+        private static object TextJoin(List<Expression> p)
+        {
+            List<string> values = new List<string>();
+            var p0 = p[0].Evaluate();
```

Comment:
We should prefer `var delimiter = (string)p[0];`

The `Expression.cs` file contains overloads for implicit casting that takes into account all the data types. The fix in #1035 is however required for this.